File: jungu/Data structure/10820.cpp
#include <iostream>
#include <string>

using namespace std;

int main() {

	string s;

	while (getline(cin, s)) {

		int a = 0;
		int A = 0;
		int number = 0;
		int space = 0;

		for (int i = 0; i < s.size(); i++) {

			if (s[i] >= 'a' && s[i] <= 'z') {
				a += 1;
			}

			else if (s[i] >= 'A' && s[i] <= 'Z') {
				A += 1;
			}

			else if (s[i] >= '0' && s[i] <= '9') {
				number += 1;
			}

			else if (s[i] == ' ') {
				space += 1;
			}
		}

		printf("%d %d %d %d\n", a, A, number, space);

	}
}

File: yoh/lec02/10820.py
a = ord('a')
z = ord('z')
A = ord('A')
Z = ord('Z')
zero = ord('0')
nine = ord('9')
while True:
	aString = raw_input()
	if aString == '':
		break

	lower, upper, number, space = 0, 0, 0, 0
	for letter in aString:
		letter_ascii = ord(letter)
		if a <= letter_ascii <= z:
			lower += 1
		elif A <= letter_ascii <= Z:
			upper += 1
		elif zero <= letter_ascii <= nine:
			number += 1
		elif letter_ascii == ord(' '):
			space += 1
	print lower, upper, number, space